File: problems/1629-slowest-key/solution.py
class Solution:
    def slowestKey(self, releaseTimes: List[int], keysPressed: str) -> str:
        max_duration = releaseTimes[0]
        slowest_key = keysPressed[0]

        for i in range(1, len(releaseTimes)):
            duration = releaseTimes[i] - releaseTimes[i-1]
            if duration > max_duration:
                max_duration = duration
                slowest_key = keysPressed[i]
            elif duration == max_duration:
                if keysPressed[i] > slowest_key:
                    slowest_key = keysPressed[i]

        return slowest_key

File: problems/1629-slowest-key/solution.java
class Solution {
    public char slowestKey(int[] releaseTimes, String keysPressed) {
        int n = releaseTimes.length;
        char slowest = keysPressed.charAt(0);
        int maxDuration = releaseTimes[0];

        for (int i = 1; i < n; i++) {
            int duration = releaseTimes[i] - releaseTimes[i - 1];
            if (duration > maxDuration) {
                maxDuration = duration;
                slowest = keysPressed.charAt(i);
            } else if (duration == maxDuration) {
                slowest = (char) Math.max(slowest, keysPressed.charAt(i));
            }
        }

        return slowest;
    }
}